test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DollarApplyGroupByTests.cs

```diff
@@ -0,0 +1,1617 @@
+            Assert.Equal(464.72, singleResult.SumDoubleProp);
+            Assert.Equal(534.02, singleResult.SumNullableDoubleProp);
+            Assert.Equal(559.4M, singleResult.SumDecimalProp);
+            Assert.Equal(393.7M, singleResult.SumNullableDecimalProp);
+            Assert.Equal(1298L, singleResult.SumLongProp);
+            Assert.Equal(993L, singleResult.SumNullableLongProp);
+            Assert.Equal(333.79f, singleResult.SumSingleProp);
+            Assert.Equal(528.44f, singleResult.SumNullableSingleProp);
+        }
+
+        [Fact]
+        public void GroupByResultSelector_Sum_BySingleProperty()
+        {
+            var queryable = this.dsContext.CreateQuery<Number>(numbersEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => d1.RowParity, (d2, d3) => new
+            {
+                RowParity = d2,
+                SumIntProp = d3.Sum(d4 => d4.IntProp),
+                SumNullableIntProp = d3.Sum(d4 => d4.NullableIntProp),
+                SumDoubleProp = d3.Sum(d4 => d4.DoubleProp),
+                SumNullableDoubleProp = d3.Sum(d4 => d4.NullableDoubleProp),
+                SumDecimalProp = d3.Sum(d4 => d4.DecimalProp),
+                SumNullableDecimalProp = d3.Sum(d4 => d4.NullableDecimalProp),
+                SumLongProp = d3.Sum(d4 => d4.LongProp),
+                SumNullableLongProp = d3.Sum(d4 => d4.NullableLongProp),
+                SumSingleProp = d3.Sum(d4 => d4.SingleProp),
+                SumNullableSingleProp = d3.Sum(d4 => d4.NullableSingleProp),
+            });
+
+            Assert.Equal(
+                string.Format(
+                    "{0}/Numbers?$apply=groupby((RowParity),aggregate(" +
+                    "IntProp with sum as SumIntProp,NullableIntProp with sum as SumNullableIntProp," +
+                    "DoubleProp with sum as SumDoubleProp,NullableDoubleProp with sum as SumNullableDoubleProp," +
+                    "DecimalProp with sum as SumDecimalProp,NullableDecimalProp with sum as SumNullableDecimalProp," +
+                    "LongProp with sum as SumLongProp,NullableLongProp with sum as SumNullableLongProp," +
+                    "SingleProp with sum as SumSingleProp,NullableSingleProp with sum as SumNullableSingleProp))", serviceUri),
+                aggregateQuery.ToString()
+            );
+
+            MockGroupBy_Sum_BySingleProperty();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Equal(2, aggregateResult.Length);
+            Assert.Equal("Even", aggregateResult[0].RowParity);
+            Assert.Equal("Odd", aggregateResult[1].RowParity);
+        }
+
+        [Fact]
+        public void GroupByResultSelector_Sum_ByMultipleProperties()
+        {
+            var queryable = this.dsContext.CreateQuery<Number>(numbersEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => new { d1.RowParity, d1.RowCategory }, (d2, d3) => new

```

Comment:
>new { d1.RowParity, d1.RowCategory } [](start = 57, length = 36)

can we groupby a computed property?

Reply:
@xuzhg No. We don't support that in ODL, no? I also looked through the OData specification and I couldn't find a section on grouping by computed property